redhat-developer/web-terminal-tooling
# Update the tooling to the latest versions

### What does this PR does?

This PR updates all tooling to be the latest versions.

oc: 4.5.0-rc.1 -> 4.5.3
odo:  v1.2.2 -> v1.2.4
helm: v3.1.3 -> v3.2.3
kubectx & kubens: v0.9.0 -> v0.9.1

It also fixes completion for helm

### How to test this PR?

```
cat <<EOF | oc apply -f -
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha1
metadata:
  name: web-terminal-new-tooling
  annotations:
    # it's important to make workspace immutable to make sure nobody with right access
    # won't set custom editor to steal creator's token
    controller.devfile.io/restricted-access: "true"
  labels:
    # it's a label OpenShift console uses a flag to mark terminal's workspaces
    console.openshift.io/terminal: "true"
spec:
  started: true
  routingClass: 'web-terminal'
  template:
    components:
      - plugin:
          name: web-terminal
          id: redhat-developer/web-terminal/4.5.0
      - container:
          memoryLimit: "256Mi"
          name: tooling
          image: quay.io/sleshche/web-terminal-tooling:new-tooling
          args: ["tail", "-f", "/dev/null"]
          env:
            - name: PS1
              value: \[\e[34m\]>\[\e[m\]\[\e[33m\]>\[\e[m\]
EOF

oc get pod
oc port-forward workspace1ba0a7a6015440db-64dc7b6d9d-6gbs4 4444

curl --header "Content-Type: application/json" \
 --header "X-Access-Token: $(oc whoami -t)" \
  --request POST \
  --data {} \
  -k \
    https://localhost:4444/exec/init

kc exec -ti workspace1ba0a7a6015440db-64dc7b6d9d-6gbs4 -c tooling bash
# make sure tooling with completion works as expected
```
